### PR TITLE
Layer manager

### PR DESCRIPTION
**Pull request by pgiraud**

### CODE REVIEW

File: geoportailv3/static/js/layermanager/layermanager.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<div ng-if="!layermanagerCtrl.layers.length" translate>No layer selected</div>
+<ul id="layermanager-{{::layermanagerCtrl.uid}}" class="layermanager">
+  <li ng-repeat="layer in layermanagerCtrl.layers" class="panel">
+    {{ layer.get('label') }}
+    <a data-toggle="collapse" data-parent="#layermanager-{{::layermanagerCtrl.uid}}" href="#layermanager-item-{{ $index }}">
```
Comment (elemoine) [inline geoportailv3/static/js/layermanager/layermanager.html]: You sometimes use no spaces within `{{ }}` and sometimes you do.

Comment (elemoine) [inline geoportailv3/static/js/layermanager/layermanager.html]: And FWIW I don't think we need spaces.

File: geoportailv3/static/js/layermanager/layermanager.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      <span class="collapsed fa fa-ellipsis-h"></span>
+      <span class="expanded fa fa-remove"></span>
+    </a>
+    <div id="layermanager-item-{{ $index }}" class="collapse">
```
Comment (elemoine) [inline geoportailv3/static/js/layermanager/layermanager.html]: Shouldn't we use the uid trick here again?

File: geoportailv3/static/js/layermanager/layermanagerdirective.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ */
+app.layermanagerDirective = function(appLayermanagerTemplateUrl) {
+  return {
+    restrict: 'E',
+    scope: {
+      'map': '=appLayermanagerMap',
+      'layers': '=appLayermanagerLayers'
+    },
+    controller: 'AppLayermanagerController',
+    controllerAs: 'layermanagerCtrl',
```
Comment (elemoine) [inline geoportailv3/static/js/layermanager/layermanagerdirective.js]: I don't think we need "complex" name for directive controllers. The name is local to the directive, so I think that we can just use "ctrl".

File: geoportailv3/templates/index.html [hunk 4]
```diff
@@ -51,18 +51,25 @@ <h2 translate>Layers</h2>
             ✕
           </button>
           <ul class="nav nav-tabs nav-justified">
-            <li role="presentation">
-              <a href="#mylayers" ng-click="mainCtrl.showTab($event);">My Layers</a>
-            </li>
             <li role="presentation" class="active">
-              <a href="#catalog" ng-click="mainCtrl.showTab($event);">Catalog</a>
+              <a href="#mylayers" data-toggle="tab">My Layers
+                <span ng-if="mainCtrl.selectedLayers.length > 0">({{ mainCtrl.selectedLayers.length }})</span>
```
Comment (elemoine) [inline geoportailv3/templates/index.html]: You use spaces after `{{` and before `}}` here again.

File: geoportailv3/static/js/maincontroller.js [hunk 5]
```diff
@@ -86,6 +90,16 @@ app.MainController.prototype.setMap_ = function() {
 
 
 /**
+ * @private
+ */
+app.MainController.prototype.updateSelectedLayers_ = function() {
+  var items_ = this['map'].getLayers().getArray().slice().reverse();
+  items_.pop();
+  this['selectedLayers'] = items_;
```
Comment (elemoine) [inline geoportailv3/static/js/maincontroller.js]: This code deserves a comment (in the code) I think. The call to `pop` is obscure if you don't know that we want to exclude the background layers here.

Comment (elemoine) [inline geoportailv3/static/js/maincontroller.js]: I also note that we will need a more complex filter when we'll use vector layers for drawing etc.

File: geoportailv3/static/js/layermanager/layermanager.html (hunk 2 above)
Comment (pgiraud) [inline geoportailv3/static/js/layermanager/layermanager.html]: No. I don't think it's worth since it's a flat array.

**Comment by pgiraud** [comment]
I took Éric's comments into account. I'm now waiting for travis tests to pass and will merge if everything goes well.

**Pull request merged**